packages/react/src/experimental/AiChat/OneSwitch.tsx

```diff
@@ -24,22 +26,32 @@ export const OneSwitch = ({
         checked={open}
         aria-label={open ? translations.ai.closeChat : translations.ai.openChat}
         className={cn(
-          "group relative h-8 w-12 rounded-full border-none bg-f1-background-inverse-secondary shadow-[0_2px_6px_-1px_rgba(13,22,37,.04),inset_0_-2px_4px_rgba(13,22,37,.04)] hover:bg-f1-background-hover",
-          "after:pointer-events-none after:absolute after:inset-0 after:rounded-full after:ring-1 after:ring-inset after:ring-f1-border after:transition-all after:content-['']",
-          "data-[state=checked]:bg-gradient-to-r data-[state=checked]:from-[hsla(229,57%,76%,0.7)] data-[state=checked]:via-[hsla(348,80%,50%,0.7)] data-[state=checked]:to-[hsla(18,80%,50%,0.7)]",
+          "group relative h-8 w-12 rounded-full border-none bg-f1-background-inverse-secondary transition-all hover:bg-f1-background-hover",
+          "shadow-[0_2px_6px_-1px_rgba(13,22,37,.04),inset_0_0_4px_rgba(13,22,37,.04)] data-[state=checked]:shadow-[0_2px_6px_-1px_rgba(13,22,37,.04),inset_0_0_4px_rgba(13,22,37,.6)]",
+          "after:pointer-events-none after:absolute after:inset-0 after:rounded-full after:ring-1 after:ring-inset after:ring-f1-border after:transition-all after:content-[''] data-[state=checked]:after:ring-f1-border-inverse",
+          "before:absolute before:inset-0 before:rounded-full before:bg-gradient-to-r before:from-[hsla(229,57%,76%,0.7)] before:via-[hsla(348,80%,50%,0.7)] before:to-[hsla(18,80%,50%,0.7)] before:opacity-0 before:transition-all before:duration-300 before:content-[''] data-[state=checked]:before:opacity-90",
           disabled && "cursor-not-allowed opacity-50",
           focusRing(),
           className
         )}
         disabled={disabled}
+        onMouseEnter={() => setIsHover(true)}
+        onMouseLeave={() => setIsHover(false)}
       >
         <SwitchPrimitive.Thumb
           className={cn(
-            "block h-[1.375rem] w-[1.375rem] translate-x-[0.25rem] rounded-full bg-f1-background transition-transform duration-300 group-hover:bg-gradient-to-r group-hover:from-[hsl(229,57%,76%)] group-hover:via-[hsl(348,80%,50%)] group-hover:to-[hsl(18,80%,50%)] *:group-hover:hidden data-[state=checked]:translate-x-[1.25rem] data-[state=checked]:bg-transparent"
+            "block h-[1.375rem] w-[1.375rem] translate-x-[5px] rounded-full transition-transform duration-300 data-[state=checked]:translate-x-[21px]"
```

Comment:
@dani-moreno why did you switch to pixels? height and widths are in rems, they could change. I think the translate values should also be relative units

Reply:
chatted in Slack. We decided to convert it back to rems